bip-tx-terminology.mediawiki

```diff
@@ -0,0 +1,286 @@
+
+; Sighash Flag
+: [TC] Part of the Signature. A one-byte field that indicates which parts of the transaction a signature commits to.
+
+; Taproot Branch
+: [TC] Part of a Control Block that specifies the hashing partners necessary to prove the existence of a Leaf Script in a Script Tree.
+
+; Taproot Internal Key
+: [TC] (aka Inner Key). The untweaked public key before it is tweaked with the Merkle root of the Script Tree to create the Taproot Output Key. Appears as part of a Control Block for a script path P2TR input.
+
+; Taproot Output Key
+: [TC] The public key present in the Witness Program of a P2TR output script composed by tweaking the Inner Key with the Script Tree's Merkle root.
+
+; Tapscript
+: [AT] The variant of the Script language used in P2TR Leaf Scripts (see BIP342).
+
+; Transaction Header
+: [AT] Collective term for the serialization artifacts and transaction fields that appear only once in the transaction serialization regardless of lengths of the Input and Output Lists: Transaction Version, Marker (segwit only), Flag (segwit only), length of Transaction Input List, length of Transaction Output List, Locktime
+
+; Transaction Input
+: [TC] aka Input. An element of the input list of the transaction. Must include an outpoint, an input script, and a sequence. The input script can have various content, corresponding to the UTXO that is being spent. The input script has length 0 for native segwit inputs. In segwit transactions, every input must be accompanied by a Witness Stack. The Witness Stack is not part of the Transaction Input.
+
+; Transaction Input List
+: [AT] The enumeration of all Transaction Inputs of a transaction.
+
+; Transaction Output
+: [TC] Creates a UTXO by specifying an amount and an output script.
+
+; Transaction Output List
+: [AT] The enumeration of all Transaction Outputs of a transaction
+
+; txid
+: [TC] Part of an Outpoint that identifies the transaction that created a UTXO. Result of hashing a transaction without its witness structure.
+
+; Version
+: [TC] Part of the transaction header that specifies the transaction version.  Serializes as the four byte <code>nVersion</code> integer field.
+
+; Witness Item
+: [TC] An element of a witness stack. Witness Items are: Witness Script (P2WSH only), Script Arguments, Leaf Script (P2TR scriptpath only), Control Block (P2TR scriptpath only), Annex (P2TR only). BIP341 also refers to Witness Items as “Witness Elements” or “Witness Stack Elements”.
+
+; Witness Program
+: [AT] A script template that forwards input validation to the Witness Stack. Witness Programs are a type of Forwarding Script. Witness Programs appear in the Output Script for native segwit outputs and in the Input Script for wrapped segwit inputs.
+
+; Witness Structure
+: [TC] The part of the serialized transaction that contains the witness stacks for each input.
```

Comment:
Since it's the part of a serialized transaction, should this be a SA instead of a TC?

Reply:
Actually, since the count of Witness Stacks is defined by the input counter, it’s literally just the concatenation of all Witness Stacks, so there are no additions to the serialization. I’ve redefined it as an [AT].